zipline/data/us_equity_pricing.py

```diff
@@ -841,14 +862,12 @@ def calc_dividend_ratios(self, dividends):
 
         daily_bar_reader = self._daily_bar_reader
 
-        calendar = self._calendar
-
         effective_dates = full(len(amounts), -1, dtype=int64)
-
+        calendar = self._calendar
         for i, amount in enumerate(amounts):
             sid = sids[i]
             ex_date = ex_dates[i]
-            day_loc = calendar.get_loc(ex_date)
+            day_loc = calendar.get_loc(ex_date, method='bfill')
```

Comment:
if you have a dividend that is payed out of a non-trading day it should use the last trading day's price